File: Python/Endpoint Examples/JSON Payload/pdf-with-added-attachment.py
import requests
import json

with open('/path/to/pdf_file', 'rb') as f:
    upload_pdf_data = f.read()

print("Uploading PDF file...")
upload_pdf_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_pdf_data,
                    headers={'Content-Type': 'application/octet-stream', 'content-filename': 'pdf_file.pdf', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Upload PDF response status code: " + str(upload_pdf_response.status_code))


with open('/path/to/file_attachment', 'rb') as f:
    upload_attachment_data = f.read()

print("Uploading file attachment...")
upload_attachment_response = requests.post(url='https://api.pdfrest.com/upload',
                    data=upload_attachment_data,
                    headers={'Content-Type': 'application/octet-stream', 'Content-Filename': 'data_file.xml', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})

print("Upload data file response status code: " + str(upload_attachment_response.status_code))

if upload_pdf_response.ok and upload_attachment_response.ok:
    upload_pdf_response_json = upload_pdf_response.json()
    print(json.dumps(upload_pdf_response_json, indent = 2))

    upload_attachment_response_json = upload_attachment_response.json()
    print(json.dumps(upload_attachment_response_json, indent = 2))


    uploaded_pdf_id = upload_pdf_response_json['files'][0]['id']
    uploaded_attachment_id = upload_attachment_response_json['files'][0]['id']
    added_attachment_data = { "id" : uploaded_pdf_id, "id_to_attach": uploaded_attachment_id }
    print(json.dumps(added_attachment_data, indent = 2))


    print("Processing file...")
    added_attachment_response = requests.post(url='https://api.pdfrest.com/pdf-with-added-attachment',
                        data=json.dumps(added_attachment_data),
                        headers={'Content-Type': 'application/json', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})



    print("Processing response status code: " + str(added_attachment_response.status_code))
    if added_attachment_response.ok:
        added_attachment_response_json = added_attachment_response.json()
        print(json.dumps(added_attachment_response_json, indent = 2))

    else:
        print(added_attachment_response.text)
else:
    print(upload_pdf_response.text)
    print(upload_attachment_response.text)

File: JavaScript/Endpoint Examples/JSON Payload/pdf-with-added-attachment.js
var axios = require("axios");
var FormData = require("form-data");
var fs = require("fs");

var pdf_upload_data = fs.createReadStream("/path/to/pdf_file");

var pdf_upload_config = {
  method: "post",
  maxBodyLength: Infinity,
  url: "https://api.pdfrest.com/upload",
  headers: {
    "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
    "Content-Filename": "pdf_filename.pdf",
    "Content-Type": "application/octet-stream",
  },
  data: pdf_upload_data, // set the data to be sent with the request
};

// send request and handle response or error
axios(pdf_upload_config)
  .then(function (pdf_upload_response) {
    var pdf_uploaded_id = pdf_upload_response.data.files[0].id;

    var attachment_upload_data = fs.createReadStream(
      "/path/to/attachment_file"
    );

    var attachment_upload_config = {
      method: "post",
      maxBodyLength: Infinity,
      url: "https://api.pdfrest.com/upload",
      headers: {
        "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
        "Content-Filename": "attachment_filename.xml",
        "Content-Type": "application/octet-stream",
      },
      data: attachment_upload_data, // set the data to be sent with the request
    };

    axios(attachment_upload_config)
      .then(function (attachment_upload_response) {
        console.log(JSON.stringify(attachment_upload_response.data));
        var attachment_uploaded_id =
          attachment_upload_response.data.files[0].id;

        var attach_config = {
          method: "post",
          maxBodyLength: Infinity,
          url: "https://api.pdfrest.com/pdf-with-added-attachment",
          headers: {
            "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
            "Content-Type": "application/json",
          },
          data: {
            id: pdf_uploaded_id,
            id_to_attach: attachment_uploaded_id,
          }, // set the data to be sent with the request
        };

        // send request and handle response or error
        axios(attach_config)
          .then(function (attach_response) {
            console.log(JSON.stringify(attach_response.data));
          })
          .catch(function (error) {
            console.log(error);
          });
      })
      .catch(function (error) {
        console.log(error);
      });
  })
  .catch(function (error) {
    console.log(error);
  });
